pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -268,60 +268,76 @@ func (state PipelineRunState) GetChildReferences(taskRunVersion string, runVersi
 		if !rprt.CustomTask && rprt.TaskRun == nil && rprt.ResolvedConditionChecks == nil {
 			continue
 		}
+		var childRef v1beta1.ChildStatusReference
+		if rprt.CustomTask {
+			childRef = rprt.getChildRefForRun(runVersion)
+		} else {
+			childRef = rprt.getChildRefForTaskRun(taskRunVersion)
+		}
+		childRefs = append(childRefs, childRef)
+	}
+	return childRefs
+}
 
-		var childAPIVersion string
-		var childTaskKind string
-		var childName string
-		var childConditions []*v1beta1.PipelineRunChildConditionCheckStatus
+func (rprt *ResolvedPipelineRunTask) getChildRefForRun(runVersion string) v1beta1.ChildStatusReference {
+	childName := rprt.RunName
+	childTaskKind := "Run"
```

Comment:
suggest using const if does not exist already, I think there should be a const already